lib/std/json/dynamic_test.zig

```diff
@@ -352,6 +352,17 @@ test "many object keys" {
     try testing.expectEqualStrings("v5", parsed.value.object.get("k5").?.string);
 }
 
+test "negative zero" {
+    const doc = "-0";
+    var fbs = std.io.fixedBufferStream(doc);
+    var reader = smallBufferJsonReader(testing.allocator, fbs.reader());
+    defer reader.deinit();
+    var parsed = try parseFromTokenSource(Value, testing.allocator, &reader, .{});
+    defer parsed.deinit();
+
+    try testing.expect(@as(u64, @bitCast(parsed.value.float)) == @as(u64, 0x8000000000000000));
```

Comment:
:100: for tests.

Is there really no stdlib function to test if a float is -0? if we have a use case for it here, we might want to consider putting a function in a more general purpose location. However, we don't need to block this PR for factoring out a function. (The `Requested changes` are for the other comment, not this one.)

Reply:
No, but I agree that it would be useful to have one! For now I have changed this to `parsed.value.float == 0 and std.math.signbit(parsed.value.float)`, which should be clearer to understand than the bit cast.

Reply:
Added one in https://github.com/ziglang/zig/pull/17740, depending on which PR gets merged first I'll either update this one to use it or fix up the tests in the other one.